### PR TITLE
Remove batch size bytes limits

### PR DESCRIPTION
**Pull request by zpuller**
Testing with higher batch sizes has shown potential performance improvements in certain scenarios, and has not shown any evidence of query failures or data corruption, so we think we are comfortable removing the existing limits we've been imposing on these configs.

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala [hunk 1]
```diff
@@ -223,8 +223,6 @@ case class RequireSingleBatchWithFilter(filterExpression: GpuExpression)
 case class TargetSize(override val targetSizeBytes: Long)
```
Comment (zpuller) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala:223]: not sure if we still need `targetSizeBytes`

Comment (zpuller) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala:223]: I think `CoalesceSizeGoal` still needs it

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala [hunk 2]
```diff
@@ -557,8 +557,6 @@ val GPU_COREDUMP_PIPE_PATTERN = conf("spark.rapids.gpu.coreDump.pipePattern")
       "cudf row count limit of a column.")
     .commonlyUsed()
     .bytesConf(ByteUnit.BYTE)
-    .checkValue(v => v >= 0 && v <= Integer.MAX_VALUE,
```
Comment (gerashegalov) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala:560]: The limit is still reflected in the `.doc(...)` on L556

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala (hunk 1 above)
Comment (jlowe) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala:223]: Yes, we still need targetSizeBytes.

**Review by abellina — APPROVED** [review]

**Comment by abellina** [comment]
build

**Pull request merged**